src/pages/Admin/Room/Edit/RoomEdit.tsx

```diff
@@ -134,24 +54,96 @@ const EditRoom: React.FC<EditRoomProps> = ({
 
       <Modal.Body>
         <Form onSubmit={handleSubmit(onSubmit)}>
-          {renderNameInput()}
-          {renderUrlInput()}
-          {renderDescriptionInput()}
-          {renderLogoInput()}
-
-          {roomTemplate?.customInputs?.map((input) => renderCustomInput(input))}
-
-          {renderTemplateSelect()}
+          <Form.Row>
+            <div className="input-wrapper">
+              <Form.Label>Name your room</Form.Label>
+              <Form.Control
+                type="text"
+                ref={register}
+                name="title"
+                placeholder="Room name"
+                custom
+              />
+            </div>
+          </Form.Row>
+
+          <Form.Row>
+            <div className="input-wrapper">
+              <Form.Label>The room url</Form.Label>
+              <Form.Control
+                type="text"
+                ref={register}
+                name="url"
+                placeholder="Room url"
+                custom
+              />
+            </div>
+          </Form.Row>
+
+          <Form.Row>
+            <div className="input-wrapper">
+              <Form.Label>The room description (optional)</Form.Label>
+              <Form.Control
+                type="text"
+                ref={register}
+                name="description"
+                placeholder="Description"
+                custom
+              />
+            </div>
+          </Form.Row>
+
+          <Form.Row>
+            <div className="input-wrapper">
+              <Form.Label>
+                How you want the room to appear on the map
+              </Form.Label>
+              <ImageInput
+                onChange={handleImageChange}
+                name="image"
+                forwardRef={register}
+                small
+                nameWithUnderscore
+                imgUrl={room.image_url}
+              />
+            </div>
+          </Form.Row>
+
+          {roomTemplate?.customInputs?.map((input) => (
+            <div className="input-wrapper" key={input.name}>
+              <span>{input.title}</span>
+              <input type="text" name={input.name} ref={register} />
+            </div>
+          ))}
 
-          <S.ButtonsWrapper>
+          <Form.Row>
+            <div className="input-wrapper">
+              <Form.Label>Change room template</Form.Label>
+              <Form.Control
+                as="select"
+                custom
+                name="template"
+                ref={register}
+                onChange={(e: React.ChangeEvent<HTMLSelectElement>) =>
+                  setSelectedTemplate(e.target.value)
+                }
+              >
+                {ROOM_TEMPLATES.map((template) => (
+                  <option key={template.name}>{template.template}</option>
+                ))}
```

Comment:
Potentially this `.map` should be `useMemo`'d to prevent needing to be run on every render of this component.

Since it's operating on a constant, it doesn't even need to happen within this component. So you could have it happen once outside of this component, then just render the result here.

Reply:
Is `template.name` unique? We'd want to use the most unique `id` type value for the `key` here.